### ISSUE TITLE
Training loss is large using the strongerbasline

### ISSUE THREAD

**Issue by GuangxingHan** [JDAI-CV/fast-reid#158]
Do you have any suggestions to run the strongerbasline? Currently I am using the default params in configs/Market1501/sbs_R50.yml for training and the loss is extremely large. The test accuracy is also low.

[07/08 18:17:09] fastreid.utils.events INFO: eta: 0:55:44  iter: 2199  total_loss: 35.248  loss_cls: 30.117  loss_triplet: 4.223  time: 0.4037  data_time: 0.0019  lr: 3.50e-04  max_mem: 0M
[07/08 18:18:24] fastreid.utils.events INFO: eta: 0:54:42  iter: 2399  total_loss: 26.057  loss_cls: 23.764  loss_triplet: 2.216  time: 0.4011  data_time: 0.0020  lr: 3.50e-04  max_mem: 0M
[07/08 18:19:39] fastreid.utils.events INFO: eta: 0:58:53  iter: 2599  total_loss: 22.484  loss_cls: 21.132  loss_triplet: 1.582  time: 0.3992  data_time: 0.0019  lr: 3.50e-04  max_mem: 0M
[07/08 18:20:54] fastreid.utils.events INFO: eta: 0:57:56  iter: 2799  total_loss: 20.920  loss_cls: 19.320  loss_triplet: 1.427  time: 0.3974  data_time: 0.0019  lr: 3.50e-04  max_mem: 0M

**Comment by L1aoXingyu** [comment]
Which task are you trying?

**Comment by GuangxingHan** [comment]
> Which task are you trying?

I am just training the reid model via supervised learning on market1501 dataset.

**Comment by L1aoXingyu** [comment]
Please provide your config file

**Comment by GuangxingHan** [comment]
> Please provide your config file.

I use the config file fast-reid/configs/Market1501/sbs_R50.yml
`_BASE_: "../Base-Strongerbaseline.yml"

DATASETS:
  NAMES: ("Market1501",)
  TESTS: ("Market1501",)

OUTPUT_DIR: "logs/market1501/sbs_R50"
`

For the Base-Strongerbaseline.yml, it is defined at fast-reid/configs/Base-Strongerbaseline.yml
`_BASE_: "Base-bagtricks.yml"

MODEL:
  FREEZE_LAYERS: ["backbone"]

  BACKBONE:
    WITH_NL: True

  HEADS:
    NECK_FEAT: "after"
    POOL_LAYER: "gempool"
    CLS_LAYER: "circle"
    SCALE: 64
    MARGIN: 0.35

  LOSSES:
    NAME: ("CrossEntropyLoss", "TripletLoss",)
    CE:
      EPSILON: 0.1
      SCALE: 1.0
    TRI:
      MARGIN: 0.0
      HARD_MINING: True
      NORM_FEAT: False
      SCALE: 1.0

INPUT:
  SIZE_TRAIN: [384, 128]
  SIZE_TEST: [384, 128]
  DO_AUTOAUG: True

DATALOADER:
  NUM_INSTANCE: 16

SOLVER:
  OPT: "Adam"
  MAX_ITER: 60
  BASE_LR: 0.00035
  BIAS_LR_FACTOR: 1.
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0005
  IMS_PER_BATCH: 64

  SCHED: "WarmupCosineAnnealingLR"
  DELAY_ITERS: 30
  ETA_MIN_LR: 0.00000077

  WARMUP_FACTOR: 0.01
  WARMUP_ITERS: 10
  FREEZE_ITERS: 10

  CHECKPOINT_PERIOD: 30

TEST:
  EVAL_PERIOD: 30
  IMS_PER_BATCH: 128

CUDNN_BENCHMARK: True
`

For the Base-bagtricks.yml, it is defined at fast-reid/configs/Base-bagtricks.yml
`MODEL:
  META_ARCHITECTURE: "Baseline"

  BACKBONE:
    NAME: "build_resnet_backbone"
    NORM: "BN"
    DEPTH: 50
    LAST_STRIDE: 1
    WITH_IBN: False
    PRETRAIN: True

  HEADS:
    NAME: "BNneckHead"
    NORM: "BN"
    POOL_LAYER: "avgpool"
    NECK_FEAT: "before"
    CLS_LAYER: "linear"

  LOSSES:
    NAME: ("CrossEntropyLoss", "TripletLoss",)

    CE:
      EPSILON: 0.1
      SCALE: 1.

    TRI:
      MARGIN: 0.3
      HARD_MINING: True
      NORM_FEAT: False
      SCALE: 1.

INPUT:
  SIZE_TRAIN: [256, 128]
  SIZE_TEST: [256, 128]
  REA:
    ENABLED: True
    PROB: 0.5
    MEAN: [123.675, 116.28, 103.53]
  DO_PAD: True

DATALOADER:
  PK_SAMPLER: True
  NAIVE_WAY: True
  NUM_INSTANCE: 4
  NUM_WORKERS: 8

SOLVER:
  OPT: "Adam"
  MAX_ITER: 120
  BASE_LR: 0.00035
  BIAS_LR_FACTOR: 2.
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0005
  IMS_PER_BATCH: 64

  SCHED: "WarmupMultiStepLR"
  STEPS: [40, 90]
  GAMMA: 0.1

  WARMUP_FACTOR: 0.01
  WARMUP_ITERS: 10

  CHECKPOINT_PERIOD: 60

TEST:
  EVAL_PERIOD: 30
  IMS_PER_BATCH: 128

CUDNN_BENCHMARK: True

`

**Comment by L1aoXingyu** [comment]
Are you training with 1 GPU or multi GPUs?

**Comment by GuangxingHan** [comment]
> Are you training with 1 GPU or multi GPUs?

1 GPU

**Comment by L1aoXingyu** [comment]
There will be a `config.yaml` file in your saving model path, please show me that file.

**Comment by GuangxingHan** [comment]
> There will be a `config.yaml` file in your saving model path, please show me that file.

Thanks much for spending time on my problem. It is defined as follows:

CUDNN_BENCHMARK: true
DATALOADER:
  NAIVE_WAY: true
  NUM_INSTANCE: 16
  NUM_WORKERS: 8
  PK_SAMPLER: true
DATASETS:
  COMBINEALL: false
  NAMES:
  - Market1501
  TESTS:
  - Market1501
  - DukeMTMC
  - CUHK03
INPUT:
  DO_AUGMIX: false
  DO_AUTOAUG: true
  DO_CJ: false
  DO_FLIP: true
  DO_PAD: true
  FLIP_PROB: 0.5
  PADDING: 10
  PADDING_MODE: constant
  REA:
    ENABLED: true
    MEAN:
    - 123.675
    - 116.28
    - 103.53
    PROB: 0.5
  RPT:
    ENABLED: false
    PROB: 0.5
  SIZE_TEST:
  - 384
  - 128
  SIZE_TRAIN:
  - 384
  - 128
MODEL:
  BACKBONE:
    DEPTH: 50
    LAST_STRIDE: 1
    NAME: build_resnet_backbone
    NORM: BN
    NORM_SPLIT: 1
    PRETRAIN: true
    PRETRAIN_PATH: ''
    WITH_IBN: false
    WITH_NL: true
    WITH_SE: false
  DEVICE: cuda:0
  FREEZE_LAYERS:
  - backbone
  HEADS:
    CLS_LAYER: circle
    IN_FEAT: 2048
    MARGIN: 0.35
    NAME: BNneckHead
    NECK_FEAT: after
    NORM: BN
    NORM_SPLIT: 1
    NUM_CLASSES: 0
    POOL_LAYER: gempool
    REDUCTION_DIM: 512
    SCALE: 64
  LOSSES:
    CE:
      ALPHA: 0.2
      EPSILON: 0.1
      SCALE: 1.0
    CIRCLE:
      ALPHA: 128
      MARGIN: 0.25
      SCALE: 1.0
    FL:
      ALPHA: 0.25
      GAMMA: 2
      SCALE: 1.0
    NAME:
    - CrossEntropyLoss
    - TripletLoss
    TRI:
      HARD_MINING: true
      MARGIN: 0.0
      NORM_FEAT: false
      SCALE: 1.0
  META_ARCHITECTURE: Baseline
  PIXEL_MEAN:
  - 123.675
  - 116.28
  - 103.53
  PIXEL_STD:
  - 58.395
  - 57.120000000000005
  - 57.375
  WEIGHTS: ''
OUTPUT_DIR: logs/market1501/sbs_R50
SOLVER:
  BASE_LR: 0.00035
  BIAS_LR_FACTOR: 1.0
  CHECKPOINT_PERIOD: 30
  DELAY_ITERS: 30
  ETA_MIN_LR: 7.7e-07
  FREEZE_ITERS: 10
  GAMMA: 0.1
  HEADS_LR_FACTOR: 1.0
  IMS_PER_BATCH: 64
  MAX_ITER: 60
  MOMENTUM: 0.9
  OPT: Adam
  SCHED: WarmupCosineAnnealingLR
  STEPS:
  - 40
  - 90
  SWA:
    ENABLED: false
    ETA_MIN_LR: 3.5e-06
    ITER: 10
    LR_FACTOR: 10.0
    LR_SCHED: false
    PERIOD: 2
  WARMUP_FACTOR: 0.01
  WARMUP_ITERS: 10
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0.0005
TEST:
  AQE:
    ALPHA: 3.0
    ENABLED: false
    QE_K: 5
    QE_TIME: 1
  EVAL_PERIOD: 30
  IMS_PER_BATCH: 128
  METRIC: cosine
  PRECISE_BN:
    DATASET: Market1501
    ENABLED: false
    NUM_ITER: 300
  RERANK:
    ENABLED: false
    K1: 20
    K2: 6
    LAMBDA: 0.3

**Comment by L1aoXingyu** [comment]
It seems you are testing cross-domain performance, one suggestion is to set `REA.ENABLE: False`.

**Comment by GuangxingHan** [comment]
> It seems you are testing cross-domain performance, one suggestion is to set `REA.ENABLE: False`.

Yes, this is an important trick for cross-domain testing. Thank you for pointing that out.

However, the model training loss is still very large without using REA on the labeled training set. I try to run several times and get similar large losses. Are there any params needing carefully tuning?

**Comment by L1aoXingyu** [comment]
![image](https://user-images.githubusercontent.com/19245504/87214766-23059080-c362-11ea-80f7-bf758a727f3e.png)

loss may be large, but testing performance is also improved.

**Comment by GuangxingHan** [comment]
> ![image](https://user-images.githubusercontent.com/19245504/87214766-23059080-c362-11ea-80f7-bf758a727f3e.png)
> 
> loss may be large, but testing performance is also improved.

Ok, Thanks much. I will re-run the training and see whether the results will be improved.

The results you show is lower that the paper reports (the rank-1 and mAP are 95.4 and 88.2 respectively on Market).  It seems more likely to be the results using bag of tricks of [1].

By the way, do you have cross-domain direct testing results using your stronger baseline? e.g., Market->Duke. 

[1]. Luo H, Jiang W, Gu Y, et al. A strong baseline and batch normalization neck for deep person re-identification[J]. IEEE Transactions on Multimedia, 2019.

**Comment by L1aoXingyu** [comment]
It's just a result of iter 6200, after finishing all training iters, it will come to the result of that paper reports.

You can test the cross-domain result yourself, and just remove `REA`.

**Comment by GuangxingHan** [comment]
> It's just a result of iter 6200, after finishing all training iters, it will come to the result of that paper reports.
> 
> You can test the cross-domain result yourself and just remove `REA`.

Got it. I will do that. Thanks much.
